crates/holochain/src/conductor/conductor.rs

```diff
@@ -2409,6 +2409,17 @@ impl Conductor {
         let dna_modifiers = clone_dna.dna().modifiers.clone();
         let clone_dna_hash = clone_dna.dna_hash().to_owned();
 
+        // if DNA hash of new clone cell already exists, reject as duplicate
+        if self
+            .ribosome_store
+            .share_ref(|ribosome_store| ribosome_store.get_dna_def(&clone_dna_hash))
+            .is_some()
+        {
+            return Err(ConductorError::AppError(AppError::DuplicateDnaHash(
+                clone_dna_hash,
+            )));
+        }
+
```

Comment:
I think this check is too loose. Shouldn't it be valid for two agents to be part of the same cloned DNA on the same conductor? This would prevent that. 

Additionally, it may be possible that a cloned DNA is registered but not even installed.

I think that rather you should get the list of installed cells from the conductor state.

Reply:
I'm glad you thought of this case. Agreed, I'll extend the test suite and refactor the error logic.